src/olympia/addons/utils.py

```diff
@@ -92,3 +96,43 @@ def get_creatured_ids(category, lang=None):
     random.shuffle(others)
     random.shuffle(per_locale)
     return map(int, filter(None, per_locale + others))
+
+
+def verify_mozilla_trademark(name, user):
+    skip_trademark_check = (
+        user and user.is_authenticated() and
+        user.email.endswith(amo.ALLOWED_TRADEMARK_SUBMITTING_EMAILS))
+
+    def _check(name):
+        name = name.lower()
+        contains_trademark_symbol = any(
+            symbol in name for symbol in amo.MOZILLA_TRADEMARK_SYMBOLS)
+
+        violates_trademark = (
+            contains_trademark_symbol and not
+            name.endswith(tuple(
+                'for {}'.format(symbol)
+                for symbol in amo.MOZILLA_TRADEMARK_SYMBOLS)))
+
+        if violates_trademark:
+            raise forms.ValidationError(ugettext(
+                u'Add-on names cannot contain the Mozilla or '
+                u'Firefox trademarks. These names should not be '
+                u'contained in add-on names if at all possible.'))
```

Comment:
I feel like this is redundant. The first line ```Add-on names cannot contain the Mozilla or Firefox trademarks.``` is fine. And maybe split up 'cannot' into 'can not'.

Reply:
I'll just cite http://www.write.com/writing-guides/general-writing/word-choice/cannot-versus-can-not/ on the "cannot" thing and yeah, I think I agree about the redundancy. I'll remove it

Reply:
That is a great resource!